lib/hashes.md

```diff
@@ -5,40 +5,97 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
+another thing, no matter what the two things are. Several of the sources I was looking at referred to a hash as a collection of 
+"key-value pairs" so I dug around for a description of a key-value pair that made sense to me. The one I found described the "key" 
+as the name of a container and the "value" as the contents of the container. The key 
+does not have to be a number. The key can be a number, or a letter, or a string, or any other object that Ruby has available.
+Because the variables in a hash are not indexed by number, there is no "first" one, "second" one, etc. To associate 
+items in a hash, use a "rocket" which is this: =>.
 
 # What are some examples of information that would be Hashes as opposed to some other data type?
 
-Replace me with your response...
+Anything where you have to associate 2 ruby objects: A name and an email address, a menu item and a price, a novel
+and an author, etc. 
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
+Hashes and Arrays are similar in that they are both collections of variables. Neither an Array nor a Hash has to be all numbers,
+or all strings, or all floats, etc., all can be combinations of different types of variables. They are different in that Arrays
+have an enumerated order to them and Hashes do not. To go back to the description of a key-value pair I used in the "What is a Hash" question
+with an Array the name of the container (the key) would always be a number, and the value could be a variety of things. With a Hash
+the name of the container could be any Ruby object.
 
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
+Retrieve a particular value by using its key. After many failures I did this with a simple hash in repl.it.
+#NOTE I am including all of these bits of code that I try in repl.it because I am saving these as notes pages, and
+I want to include the code in my notes pages. I am not trying to see how much code I can turn in or anything like that.
+I know you aren't looking for code.
+
+StateCaps2 = Hash.new{}
+StateCaps2 = {"Idaho"=>"Boise", "Oregon"=>"Salem", "Vermont"=>"Montpelier", "Iowa"=>"Des Moines"}
+puts StateCaps2.keys.inspect
+puts StateCaps2.values.inspect
+puts StateCaps2["Vermont"] #This returns just the capital of Vermont, which is "Montpelier.
+
 
 # How do you add information to a Hash?
 
-Replace me with your response...
+There are probably several ways to do this, but I found a Merge method that works kind of like "join".
+I made a second hash of more states and capitals, then used the merge! method to make one larger hash.
+This tacked on the new variables at visually at least is the "end" of the old hash.
+StateCaps2 = Hash.new{}
+StateCaps2 = {"Idaho"=>"Boise", "Oregon"=>"Salem", "Vermont"=>"Montpelier", "Iowa"=>"Des Moines"}
+StateCaps3 = {"New York"=>"Albany", "Texas"=>"Austin", "Kentucky"=>"Frankfort"}
+StateCaps2.merge! StateCaps3
```

Comment:
This is doing a bit more than changing the value of an element in a Hash. As you mentioned, it's _merging_ two hashes and overwriting the values of the original with any conflicting values in the new one. So while it's true that this answers the question, I would also like you to find a way to single out a specific element in a Hash and change its value alone.